### DISCUSSION TITLE
Symbols for TTS models should be defined in hparams to allow for training other languages

### DISCUSSION THREAD

**Discussion by czyzi0** [speechbrain/speechbrain#2362] [Feature Request]
### 🚀The feature

Module `text_to_sequence.py` is very English-centric because of hard-coded values for valid symbols for TTS input, e.g. for Polish language there are some important characters missing (ą, ć, ę, ł, ń, ó, ś, ź, ż) and simple conversion to ascii will not produce good results.

To solve that there would have to be an option to modify allowed symbols in a hyperparamaters file - either as choosing one of implemented languages/character sets or as providing list of allowed symbols.

### Solution outline

That change would make speechbrain framework more inclusive for researchers doing their work in languages other than English. Also if done correctly it could invite some users of other languages to extend implemented character sets for more languages.

**Comment by Adel-Moumen** [comment]
Hi @BenoitWang & @mravanelli, what do you think? :)

**Comment by BenoitWang** [comment]
Hi @czyzi0, thanks for the proposal. However, apart from the special characters, when the user wants to change the language, some special writing rules should be sometimes added too (For example, for german, all nouns should have their first letter capitalized). And these rules are not likely to be put into the .yaml file. So the idea is that the users may change the characters together with the rules in text_to_sequence.py while adapting the model to another language.
